pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator.go

```diff
@@ -111,25 +125,29 @@ func (mutator *VMsMutator) setDefaultMachineType(vm *v1.VirtualMachine) {
 		vm.Spec.Template.Spec.Domain.Machine = &v1.Machine{}
 	}
 
-	vm.Spec.Template.Spec.Domain.Machine.Type = mutator.getPreferenceMachineType(vm)
+	if preferenceSpec != nil && preferenceSpec.Machine != nil {
+		vm.Spec.Template.Spec.Domain.Machine.Type = preferenceSpec.Machine.PreferredMachineType
+	}
 
 	// Only use the cluster default if the user hasn't provided a machine type or referenced a preference with PreferredMachineType
 	if vm.Spec.Template.Spec.Domain.Machine.Type == "" {
 		vm.Spec.Template.Spec.Domain.Machine.Type = mutator.ClusterConfig.GetMachineType()
 	}
 }
 
-func (mutator *VMsMutator) getPreferenceMachineType(vm *v1.VirtualMachine) string {
-	preferenceSpec, err := mutator.InstancetypeMethods.FindPreferenceSpec(vm)
-	if err != nil {
-		// Log but ultimately swallow any preference lookup errors here and let the validating webhook handle them
-		log.Log.Reason(err).Error("Ignoring error attempting to lookup PreferredMachineType.")
-		return ""
+func (mutator *VMsMutator) setPreferenceStorageClassName(vm *v1.VirtualMachine, preferenceSpec *v1alpha2.VirtualMachinePreferenceSpec) {
+	// Nothing to do, let's the validating webhook fail later
+	if vm.Spec.Template == nil {
```

Comment:
As above in `setDefaultMachineType` `preferenceSpec` can also be `nil` here and you should also ensure `.Volumes` isn't `nil`.